crates/abi/src/lib.rs

```diff
@@ -39,6 +39,17 @@ pub fn failure() -> ! {
     }
 }
 
+#[inline]
+pub unsafe fn miralis_end_benchmark() -> ! {
```

Comment:
A few last details on this function:
- You can make it safe, it can't cause undefined behavior or doesn't have any pre-conditions that needs to be enforced by the caller.
- Add a small doc comment too, it's a public function in a library so it's always a good practice to have one.
- Instead of having inline assembly you can use the `miralis_ecall` helper, like `success` and `failure` do (`unsafe { miralis_ecall(abi::MIRALIS_FAILURE_FID).ok() };` but with the right FID constant)

After that I think we are good :)